applications/mlflow/Taskfile.yml

```diff
@@ -0,0 +1,804 @@
+    deps: [setup:namespaces]
+    cmds:
+      - echo "Running KOTS installation test..."
+      - |
+        if [ -z "$REPLICATED_LICENSE_ID" ]; then
+          echo "ERROR: REPLICATED_LICENSE_ID environment variable must be set"
+          exit 1
+        fi
+
+        if [ -z "$REPLICATED_APP" ]; then
+          echo "ERROR: REPLICATED_APP environment variable must be set"
+          exit 1
+        fi
+
+        if [ -z "$REPLICATED_CHANNEL" ]; then
+          echo "ERROR: REPLICATED_CHANNEL environment variable must be set"
+          exit 1
+        fi
+        
+        # Create directory for license file if it doesn't exist
+        mkdir -p /tmp/replicated
+        LICENSE_FILE="/tmp/replicated/license.yaml"
+        
+        # Validate license file exists and has content
+        if [ ! -f "$LICENSE_FILE" ] || [ ! -s "$LICENSE_FILE" ]; then
+          echo "ERROR: License file does not exist or is empty at $LICENSE_FILE"
+          echo "Please download the license file using the customer:download-license task first"
+          exit 1
+        fi
+        
+        # Verify license file is valid YAML
+        if command -v yq >/dev/null 2>&1; then
+          echo "Validating license file is proper YAML before installation..."
+          if ! yq eval . "$LICENSE_FILE" > /dev/null 2>&1; then
+            echo "ERROR: License file is not valid YAML"
+            echo "License file content:"
+            cat "$LICENSE_FILE" | head -n 10
+            exit 1
+          else
+            echo "✅ License file is valid YAML"
+          fi
+        else
+          echo "WARNING: yq not found, skipping YAML validation"
+        fi
+        
+        echo "Installing latest KOTS version..."
+        curl https://kots.io/install | bash
+        
+        echo "License file at $LICENSE_FILE (first 5 lines):"
+        head -n 5 "$LICENSE_FILE"
+        
+        echo "Installing application from Replicated..."
+        echo "App: $REPLICATED_APP"
+        echo "Channel: $REPLICATED_CHANNEL"
+        echo "Using license file: $LICENSE_FILE"
+        
+        # Run KOTS install with detailed output
+        set -x
+        kubectl kots install $REPLICATED_APP/$REPLICATED_CHANNEL \
+          --shared-password=replicatedmlflow \
+          --license-file="$LICENSE_FILE" \
+          --namespace=default \
+          --wait-duration=10m \
+          --skip-preflights
+        set +x
+        
+        # Check if installation succeeded
+        if [ $? -ne 0 ]; then
+          echo "❌ KOTS installation failed"
+          echo "Checking app status:"
+          kubectl get app -n default
+          echo "Checking pods:"
+          kubectl get pods -n default
+          echo "Checking pod logs:"
+          kubectl logs -n default -l app=kotsadm --tail=50
+          exit 1
+        fi
+        
+        echo "✅ KOTS installation completed. Setting up port forwarding for testing..."
+      - task: forward:port
+
+  # Port forwarding task
+  forward:port:
```

Comment:
This is a really big task.
I wonder if you would be better served to split of the status checking into a utility task. Something that you could re-use in other tasks as well if you're just 'waiting for deploy' or something.